src/icalendar/alarms.py

```diff
@@ -178,7 +177,7 @@ class Alarms:
     This is not implemented yet.
     """
 
-    def __init__(self, component: Optional[Alarm | Event | Todo] = None):
+    def __init__(self, component: Optional[Alarm | Parent] = None):
```

Comment:
To go along with my previous comment, if `TYPE_CHECKING` is `False`, then this won't work for `Event` or `Todo`.